OptionsResolver/ParameterOptionsResolver.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Metglobal\DTOBundle\OptionsResolver;
+
+use Metglobal\DTOBundle\DTOParamConverter;
+
+class ParameterOptionsResolver extends \Symfony\Component\OptionsResolver\OptionsResolver
+{
+    public function __construct()
+    {
+        $this->setDefaults(
+            [
+                DTOParamConverter::PROPERTY_OPTION_TYPE => DTOParamConverter::DEFAULT_OPTION_TYPE,
+                DTOParamConverter::PROPERTY_OPTION_SCOPE => DTOParamConverter::DEFAULT_OPTION_SCOPE,
+                DTOParamConverter::PROPERTY_OPTION_NULLABLE => DTOParamConverter::DEFAULT_OPTION_NULLABLE,
+                DTOParamConverter::PROPERTY_OPTION_DISABLED => DTOParamConverter::DEFAULT_OPTION_DISABLED,
+                DTOParamConverter::PROPERTY_OPTION_OPTIONS => DTOParamConverter::DEFAULT_OPTION_OPTIONS,
+            ]
+        );
+        $this->setAllowedTypes(
+            DTOParamConverter::PROPERTY_OPTION_NULLABLE,
+            ['boolean', 'null']
+        );
+        
+        $this->setAllowedTypes(
+            DTOParamConverter::PROPERTY_OPTION_DISABLED,
+            ['boolean', 'null']
+        );
+        
+        $this->setAllowedTypes(
+            DTOParamConverter::PROPERTY_OPTION_OPTIONS,
+            ['array']
+        );
+        
+        $this->setAllowedValues(
+            DTOParamConverter::PROPERTY_OPTION_TYPE,
+            ['date', 'float', 'string', 'boolean', 'integer', 'int', null]
```

Comment:
Son olarak buraya birde `bool` ekleyelim ben bazen `@Parameter(type=”bool”)` yazdığım oluyor.

Reply:
Standart olması için "bool" eklemedim abi, alışkanlık olsun istedim

Reply:
Bu durum **int** ve **integer** için geçerli değil mi?

Reply:
Çok haklı bir istek abi. int, integer kısmını kaçırmışım şuan farkettim. Bunuda ekliyorum, teşekkürler uyardığın için

Reply:
Hatta şöyle bir öneride bulunabilirim sana:

- (int), (integer) - cast to integer
- (bool), (boolean) - cast to boolean
- (float), (double), (real) - cast to float
- (string) - cast to string
- (array) - cast to array
- (object) - cast to object
- (unset) - cast to NULL